fix Issue 14276 - DWARF debug info for SIMD broken

MartinNowak opened the pull request:
- subrange needs to follow vector type as child
- base type is encoded separately

yebblies commented on src/backend/dwarf.c:
```diff
@@ -2149,18 +2149,20 @@ unsigned dwarf_typidx(type *t)
-        case TYulong4:   tbase = tsulong;  goto Lvector;
-        case TYllong2:   tbase = tsllong;  goto Lvector;
-        case TYullong2:  tbase = tsullong; goto Lvector;
+        unsigned char dim;
```
Probably better to define this outside the switch.

MartinNowak replied:
I computed the dimension instead.

MartinNowak commented:
Disable linux-32 tests, because core.simd.int4 isn't available.
Now should pass all tests.

MartinNowak committed:
fix Issue 14276 - DWARF debug info for SIMD broken

- subrange needs to follow vector type as child
- base type is encoded separately

MartinNowak commented:
> Disable linux-32 tests, because core.simd.int4 isn't available.

Using new gdb testability and replaced `int4` with `int[4]` on non X86_64 platforms.
